go/enclave/components/batch_executor.go

```diff
@@ -722,10 +726,33 @@ func (executor *batchExecutor) verifySyntheticTransactionsSuccess(transactions c
 	return nil
 }
 
+// 28 bytes entropy || 4 bytes timestamp (50 days)
+func buildMixDigest(entropy gethcommon.Hash, timeDelta int32) []byte {
+	digest := make([]byte, 32)
+	copy(digest[:28], entropy[:28])
+	copy(digest[28:], []byte{byte(timeDelta >> 24), byte(timeDelta >> 16), byte(timeDelta >> 8), byte(timeDelta)})
+	return digest
+}
+
 func (executor *batchExecutor) executeTx(ec *BatchExecutionContext, tx *common.L2PricedTransaction, offset int, noBaseFee bool) (*core.TxExecResult, error) {
 	ethHeader := *ec.EthHeader
-	before := ethHeader.MixDigest
-	ethHeader.MixDigest = executor.entropyService.TxEntropy(before.Bytes(), offset)
+	fullEntropy := executor.entropyService.TxEntropy(ethHeader.MixDigest.Bytes(), offset)
+
+	// we use the "mixdigest" field to transport both the unique transaction entropy and the unique transaction timestamp
+
+	txTime := tx.Tx.Time().UnixMilli()
+	if tx.FromSelf {
+		txTime = int64(ethHeader.Time) * 1000
```

Comment:
this might become a problem down the line, maaaybe. Good enough for now i think.